### PR TITLE
Update Select docs to clarify value, labelKey

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
Updating Select docs to clarify intended usage of value and labelKey. 

#### Where should the reviewer start?
src/js/components/Select/doc.js

#### Do the grommet docs need to be updated?
Updated here.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Comment by britt6612** [comment]
This would close issue #5201

**Review by britt6612 — APPROVED** [review]

File: src/js/components/Select/doc.js [hunk 1]
```diff
@@ -121,7 +121,8 @@ export const doc = Select => {
       PropTypes.func,
     ]).description(
       `When the options array contains objects, this property indicates how
-      to determine the label of each option. If a string is
+      to determine the label of each option. The label is what is displayed 
+      in the Select button and its dropdown options. If a string is
```
Comment (ericsoderberghp) [inline src/js/components/Select/doc.js:125]:
```suggestion
      visually for the user, both to indicate the selected value and in the options list. If a string is
```

File: src/js/components/Select/doc.js [hunk 2]
```diff
@@ -214,10 +215,12 @@ export const doc = Select => {
           PropTypes.number,
         ]),
       ),
-    ]).description(`Currently selected value. This can be an array
-      when multiple. Passing an element allows the caller to control how
-      the value is rendered. Passing an element is deprecated. Instead,
-      use the 'valueLabel' property.`),
+    ]).description(`Currently selected value. This should be an option from
```
Comment (ericsoderberghp) [inline src/js/components/Select/doc.js:218]: It actually should be one of the `options` objects if `valueKey` has `reduce: true`. Perhaps here we can just refer to needing to match whatever the `valueKey` results in for the selected option. We might need to improve the description of `valueKey` to be clearer. I think it's so-so at the moment.

**Comment by taysea** [comment]
This is very helpful, I feel like this direction clarifies a lot. Only point of clarification I still am wondering about is, when the options are objects, should what is stored in Select's value never include the label portion of the option object? or is that up to the caller to decide. Wondering because in this example, it still works for the value to contain the label portion of the option: https://codesandbox.io/s/async-haze-9mvsx?file=/src/Select.js

**Comment by ericsoderberghp** [comment]
> This is very helpful, I feel like this direction clarifies a lot. Only point of clarification I still am wondering about is, when the options are objects, should what is stored in Select's value never include the label portion of the option object? or is that up to the caller to decide. Wondering because in this example, it still works for the value to contain the label portion of the option: https://codesandbox.io/s/async-haze-9mvsx?file=/src/Select.js

What is being stored in "selectedName" is really the "selectedOption". It is initialized with `OPTIONS[0]` and then updated with the `event.option` in `onSelect`. Since valueKey doesn't set `reduce`, then `value` should be passed one of the `options`. This is what "Otherwise, the value(s) should match the selected options." is intending to communicate.

**Comment by taysea** [comment]
> > This is very helpful, I feel like this direction clarifies a lot. Only point of clarification I still am wondering about is, when the options are objects, should what is stored in Select's value never include the label portion of the option object? or is that up to the caller to decide. Wondering because in this example, it still works for the value to contain the label portion of the option: https://codesandbox.io/s/async-haze-9mvsx?file=/src/Select.js
> 
> What is being stored in "selectedName" is really the "selectedOption". It is initialized with `OPTIONS[0]` and then updated with the `event.option` in `onSelect`. Since valueKey doesn't set `reduce`, then `value` should be passed one of the `options`. This is what "Otherwise, the value(s) should match the selected options." is intending to communicate.

Got it, I see the connection now. Maybe then something along the lines of: "Otherwise, the value should be one (or multiple, when multiple = true) of the options from the options array."

**Comment by britt6612** [comment]
@ericsoderberghp I think this is very helpful for the definitions just wondering if we can also add a few comments to the code.. I would not of known this behavior without reading this.. There are a few comments already but we can elaborate a little more to be clear.

**Comment by ericsoderberghp** [comment]
> @ericsoderberghp I think this is very helpful for the definitions just wondering if we can also add a few comments to the code.. I would not of known this behavior without reading this.. There are a few comments already but we can elaborate a little more to be clear.

What about just pasting these in the code as well, so they are available there too?

**Comment by britt6612** [comment]
> > @ericsoderberghp I think this is very helpful for the definitions just wondering if we can also add a few comments to the code.. I would not of known this behavior without reading this.. There are a few comments already but we can elaborate a little more to be clear.
> 
> What about just pasting these in the code as well, so they are available there too?

sounds good I will update this PR with those changed!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**